http-clients/src/main/java/com/palantir/remoting/http/FeignClients.java

```diff
@@ -41,8 +43,11 @@ private FeignClients() {}
     public static FeignClientFactory standard() {
         return FeignClientFactory.of(
                 new GuavaOptionalAwareContract(new JAXRSContract()),
-                new JacksonEncoder(ObjectMappers.guavaJdk7()),
-                new OptionalAwareDecoder(new TextDelegateDecoder(new JacksonDecoder(ObjectMappers.guavaJdk7()))),
+                new InputStreamDelegateEncoder(new JacksonEncoder(ObjectMappers.guavaJdk7())),
+                new OptionalAwareDecoder(
```

Comment:
Open to refactoring here. Also wondering if we still need an OptionalAwareDecoder given that we just fixed the Contract stuff.

I suppose one pitch would be we make:

     interface LimitedApplicationDecoder extends Decoder {
          boolean isSuitable(Response response, Type type)
     }

And then we can make a Decoder that takes an ordered list of Decoders to try?